packages/components/src/internal/components/search/SampleFinderSavedViewsMenu.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { FC, memo, useState, useEffect, useMemo } from 'react';
+import { Button, DropdownButton, MenuItem } from 'react-bootstrap';
+
+import { LoadingSpinner } from '../base/LoadingSpinner';
+import { SubMenuItem } from '../menus/SubMenuItem';
+
+import { ComponentsAPIWrapper, getDefaultAPIWrapper } from '../../APIWrapper';
+
+import { FinderReport } from './models';
+
+interface Props {
+    api?: ComponentsAPIWrapper;
+    loadSearch: (view: FinderReport) => any;
+    saveSearch: (saveCurrentName?: boolean) => any;
+    manageSearches: () => any;
+    currentView?: FinderReport;
+    hasUnsavedChanges?: boolean;
+    sessionViewName?: string;
+    key: any;
```

Comment:
is this type `string`?

Reply:
and is this just to trigger the `loadFinderSearches` in `useEffect`? if so, I might suggest a comment here.

Reply:
I would recommend not using `key` as a prop as this is a built-in prop for React components. When the key changes the entire component with be unmounted/mounted regardless of what is done inside the component. This likely means you don't need it here inside of `SampleFinderSavedViewsMenu`.